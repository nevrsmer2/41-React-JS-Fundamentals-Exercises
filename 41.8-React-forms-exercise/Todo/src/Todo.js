import React from 'react';

const Todo = ({ id, todo, deleteTodo }) => {
    return (
        <li>Task: {todo}
            <button onClick={() => deleteTodo(id)}>X</button>
        </li>
    );
};

export default Todo;
