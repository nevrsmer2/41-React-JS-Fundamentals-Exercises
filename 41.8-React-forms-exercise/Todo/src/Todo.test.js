import { render } from "@testing-library/react";
import React from "react";
import Todo from "./Todo";


// Smike Test
it("Todo component renders without crashing", function () {
    render(<Todo />);
});

// Snapshot Test 
it("Current structure matches structure in snapshot", function () {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

// Snapshot Test - Initial State Todo Value
it("matches snapshot", function () {
    const { asFragment } = render(<Todo todo="Wash car" />);
    expect(asFragment()).toMatchSnapshot();
});



