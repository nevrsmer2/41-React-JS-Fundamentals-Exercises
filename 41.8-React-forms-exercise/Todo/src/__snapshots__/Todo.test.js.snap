// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current structure matches structure in snapshot 1`] = `
<DocumentFragment>
  <li>
    Task: 
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li>
    Task: Wash car
    <button>
      X
    </button>
  </li>
</DocumentFragment>
`;
